nucypher/cli/deploy.py

```diff
@@ -93,131 +89,116 @@ def __init__(self):
 
 @click.command()
 @click.argument('action')
-@click.option('--contract-name', help="Deploy a single contract by name", type=click.STRING)
 @click.option('--force', is_flag=True)
+@click.option('--poa', help="Inject POA middleware", is_flag=True)
+@click.option('--no-compile', help="Inject POA middleware", is_flag=True)
+@click.option('--provider-uri', help="Blockchain provider's URI", type=click.STRING)
+@click.option('--contract-name', help="Deploy a single contract by name", type=click.STRING)
 @click.option('--deployer-address', help="Deployer's checksum address", type=EIP55_CHECKSUM_ADDRESS)
-@click.option('--registry-outfile', help="Output path for new registry", type=click.Path(), default=NodeConfiguration.REGISTRY_SOURCE)
+@click.option('--allocation-infile', help="Input path for allocation JSON file", type=EXISTING_READABLE_FILE)
 @nucypher_deployer_config
-def deploy(config,
+def deploy(click_config,
            action,
+           poa,
+           provider_uri,
            deployer_address,
            contract_name,
-           registry_outfile,
+           allocation_infile,
+           no_compile,
            force):
     """Manage contract and registry deployment"""
 
-    if not config.deployer:
-        click.secho("The --deployer flag must be used to issue the deploy command.", fg='red', bold=True)
-        raise click.Abort()
+    def __connect(deployer_address=None):
 
-    def __get_deployers():
+        # Ensure config root exists
+        if not os.path.exists(DEFAULT_CONFIG_ROOT):
+            os.makedirs(DEFAULT_CONFIG_ROOT)
 
-        config.registry_filepath = registry_outfile
-        config.connect_to_blockchain()
-        config.blockchain.interface.deployer_address = deployer_address or config.accounts[0]
-        click.confirm("Continue?", abort=True)
-        return deployers
+        # Connect to Blockchain
+        blockchain = Blockchain.connect(provider_uri=provider_uri, deployer=True, compile=not no_compile, poa=poa)
```

Comment:
Not very happy with these double negatives :(
Is there a way to convey the `no compile` meaning in an affirmative way?

Reply:
We can name the condition - `compile = not no_compile`, higher up in the file.  The Idea is that the default behaviour is to... compile.

Perhaps the flag `--disable-compilation` is better?

Reply:
I made an Issue for this since this pattern appears in several places. <https://github.com/nucypher/nucypher/issues/612>

Reply:
`ain't no compile`

Reply:
If there's an issue, I guess we can mark this as resolved.

Reply:
:heavy_check_mark: